AWSSDKSwiftCLI/Sources/AWSSDKSwiftCLI/Models/Features.swift

```diff
@@ -0,0 +1,72 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+import AWSCLIUtils
+
+struct Features: Decodable {
+    let features: [Feature]
+
+    static func fromFile(_ filePath: String) throws -> Self {
+        var fileContents: Data
+        do {
+            fileContents = try FileManager.default.loadContents(atPath: filePath)
+        } catch {
+            throw Error("Failed to load build-request.json.")
+        }
+        do {
+            return try JSONDecoder().decode(Self.self, from: fileContents)
+        } catch {
+            throw Error("Failed to decode build-request.json.")
+        }
+    }
+
+    static func mappingFromFile(_ filePath: String) throws -> [String: String] {
+        var fileContents: Data
+        do {
+            fileContents = try FileManager.default.loadContents(atPath: filePath)
+        } catch {
+            throw Error("Failed to load feature-service-id.json.")
+        }
+        do {
+            return try JSONDecoder().decode([String: String].self, from: fileContents)
+        } catch {
+            throw Error("Failed to decode feature-service-id.json.")
+        }
+    }
+}
+
+struct Feature: Decodable {
+    let featureId: String
+    let featureType: String
+    let releaseNotes: String
+
+    // Custom coding keys for decoding nested values into top level fields in Feature struct
+    private enum CodingKeys: String, CodingKey {
+        case releaseNotes
+        case featureMetadata
+
+        enum FeatureMetadataKeys: String, CodingKey {
+            case trebuchet
+
+            enum TrebuchetKeys: String, CodingKey {
+                case featureId, featureType
+            }
+        }
+    }
+
+    init(from decoder: Decoder) throws {
```

Comment:
Rather than write custom encoders/decoders, I'd just create Swift structures that match the JSON schema and let auto-Codable conformance do the hard work.  The sample JSON:
```json
    {
        "releaseNotes": "Doc update description C.",
        "featureMetadata": {
            "trebuchet": {
                "featureId": "feature-id-c",
                "featureType": "DOC_UPDATE",
            }
        }
    }
```
translates to (any of these fields can be marked optional if warranted):
```swift
struct Feature: Codable {
    let releaseNotes: String
    let featureMetadata: FeatureMetadata
    
    struct FeatureMetadata: Codable {
        let trebuchet: Trebuchet

        struct Trebuchet: Codable {
            let featureId: String
            let featureType: String
        }
    }
}
```
Swift will do all the encoding/decoding for you by generating `init(from:) throws` and `func encode(to:) throws` for each of these structs.

Reply:
Changed as suggested; regression tests pass with this change.